autoqubo/sampling_compiler.py

```diff
@@ -25,11 +25,37 @@ def _new_training_sample(input_size, idx):
             sample[i] = 1
         return sample
 
+    @staticmethod
+    def _new_test_sample(input_size):
+        sample = np.random.randint(2, size=(input_size,)).tolist()
+        return sample
+
     @staticmethod
     def _get_training_samples(input_size):
         return (SamplingCompiler._new_training_sample(input_size, idx)
                 for idx in SamplingCompiler._indices_iterator(input_size))
 
+    @staticmethod
+    def _get_test_samples(input_size, num_test_samples, training_samples):
+
+        training_samples = list(training_samples)
+
+        # Compute maximum number of testing samples and adjust
+        space_size = 2**input_size
+        max_test_samples = space_size - len(list(training_samples))
+        if num_test_samples > max_test_samples:
+            print(f"*** Warning, requested test size is {num_test_samples}, which is larger than the maximum of {max_test_samples}")
+            num_test_samples = max_test_samples
+
+        # Compute the test samples
+        test_samples = []
+        while len(test_samples) < num_test_samples:
+            sample = SamplingCompiler._new_test_sample(input_size)
+            if sample not in training_samples:
```

Comment:
I think we can remove the dependency on training_samples and check sum(sample)<3, if we assume that the training set is what we commonly use i.e. all existing solutions with 0/1/2 ones.

Reply:
OK, maybe I will leave this to you, not sure yet how to do it. For now, I will remove the testing from the portfolio optimization example (which is OK on the short term since we don't expect it to fail).